sql/sql_connect.cc

```diff
@@ -422,6 +422,7 @@ static int increment_count_by_name(const char *name, const char *role_name,
     }
   }
   user_stats->total_connections++;
+  user_stats->concurrent_connections++;
```

Comment:
There is an existing USER_CONN::connections - please check e.g. how https://github.com/facebook/mysql-5.6/commit/a0109fb6cce7ff7435f33b4b27c8552b191860b4 (grep for fill_one_user_stats) does it, and please confirm whether USER_CONN::connections is suitable for this purpose. This would also get rid of diff_disconnect